bl/evaluation/evaluator.go

```diff
@@ -107,55 +116,105 @@ func (e *Evaluator) Evaluate(policyCheckData PolicyCheckData) (PolicyCheckResult
-				if err != nil {
-					return emptyPolicyCheckResult, err
-				}
+	if !policyCheckData.IsInteractiveMode {
+		formattedResults.NonInteractiveEvaluationResults = e.formatNonInteractiveEvaluationResults(formattedResults.EvaluationResults, failedRulesByFiles, policyCheckData.PolicyName, rulesCount)
+	}
 
-				ruleSchemaJson, err := json.Marshal(ruleWithSchema.Schema)
-				if err != nil {
-					return emptyPolicyCheckResult, err
-				}
+	return PolicyCheckResultData{formattedResults, rulesData, filesData, failedRulesByFiles, rulesCount}, nil
+}
 
-				validationResult, err := jsonSchemaValidator.ValidateYamlSchema(string(ruleSchemaJson), string(configurationJson))
+func (e *Evaluator) evaluateConfiguration(failedRulesByFiles FailedRulesByFiles, policyCheckData PolicyCheckData, fileName string, configuration extractor.Configuration) error {
+	skipAnnotations := extractSkipAnnotations(configuration)
 
-				if err != nil {
-					return emptyPolicyCheckResult, err
-				}
+	configurationName, configurationKind := extractConfigurationInfo(configuration)
 
-				failedRulesByFiles = calculateFailedRulesByFiles(failedRulesByFiles, validationResult, filesConfiguration.FileName, ruleWithSchema, configurationName, configurationKind)
-			}
+	configurationJson, err := json.Marshal(configuration)
+	if err != nil {
+		return err
+	}
+
+	for _, rule := range policyCheckData.Policy.Rules {
+
+		failedRule, err := e.evaluateRule(rule, configurationJson, configurationName, configurationKind, skipAnnotations)
+		if err != nil {
+			return err
+		}
+
+		if failedRule == nil {
+			continue
 		}
 
+		addFailedRule(failedRulesByFiles, fileName, rule.RuleIdentifier, failedRule)
 	}
 
-	formattedResults := FormattedResults{}
-	formattedResults.EvaluationResults = e.formatEvaluationResults(failedRulesByFiles, len(policyCheckData.FilesConfigurations))
+	return nil
+}
 
-	if !policyCheckData.IsInteractiveMode {
-		formattedResults.NonInteractiveEvaluationResults = e.formatNonInteractiveEvaluationResults(formattedResults.EvaluationResults, failedRulesByFiles, policyCheckData.PolicyName, rulesCount)
+func (e *Evaluator) evaluateRule(rule policy_factory.RuleWithSchema, configurationJson []byte, configurationName string, configurationKind string, skipAnnotations map[string]string) (*cliClient.FailedRule, error) {
+	ruleSchemaJson, err := json.Marshal(rule.Schema)
+	if err != nil {
+		return nil, err
 	}
 
-	return PolicyCheckResultData{formattedResults, rulesData, filesData, failedRulesByFiles, rulesCount}, nil
+	validationResult, err := e.jsonSchemaValidator.ValidateYamlSchema(string(ruleSchemaJson), string(configurationJson))
+
+	if err != nil {
+		return nil, err
+	}
+
+	occurrences := countOccurrences(validationResult)
+
+	if occurrences < 1 {
+		return nil, nil
+	}
+
+	configurationData := cliClient.Configuration{
+		Name:        configurationName,
+		Kind:        configurationKind,
+		Occurrences: occurrences,
+		IsSkipped:   false,
```

Comment:
Every rule will have `IsSkipped` and `SkipMessage` by default, right? 
Is it necessary? I think lines 205-208 are enough